### ISSUE TITLE
Make all Neuron and Muscle objects Cell objects as well

### ISSUE THREAD

**Issue by gsarma** [openworm/PyOpenWorm#176]
@travs and I are working on #158 and it is not clear what Cell objects are supposed to be.  For example:
```
In [152]: c = P.Cell(name="ADAL")
In [153]: c.blast()
Out[153]: ''
```
Also:
```
In [154]: cells = P.Worm().cell()
In [155]: cells
Out[155]: ALA
```

**Comment by slarson** [comment]
Documentation is sparse, but present [over here](https://pyopenworm.readthedocs.org/en/latest/api.html#cell).  Along which lines should we be fleshing it out to add clarity?

**Comment by travs** [comment]
@slarson There are a couple separate issues here I believe, one being the lack of `Cell` objects in the database (just "ALA"), the other that the `blast` function is returning nothing. The latter is probably a result of the former.

If we agree that "`muscles + neurons == cells`" then we could change the `cells` call to something like:
```python
def cells():
    return worm.muscles().union(net.neurons())
```

Does this seem like a reasonable approach?

**Comment by slarson** [comment]
@travs  Helpful clarification.  I think the solution needs to be a bit deeper.  `Cell` is the super class of both `Muscle` and `Neuron`:

![screen shot 2015-07-06 at 4 59 59 pm](https://cloud.githubusercontent.com/assets/1037756/8536014/808d5ef6-2400-11e5-8d38-1d0ef25d59a7.png)
![screen shot 2015-07-06 at 5 00 06 pm](https://cloud.githubusercontent.com/assets/1037756/8536015/830b8572-2400-11e5-92b5-3c249ed61477.png)

What I would have expected is that all the muscles and neurons were also included as cells *in the graph*.  This is important, because any property or method that can be called on a `Cell` object, should be able to be called on a `Neuron` object.

I think we need to double check if that structure is making it into the graph or not.  As such, we should be able to query the graph to show us all Cells, and get Neurons + Muscles.  If we do it the way you suggest, I'm fearing that we are wallpapering over a potentially deeper problem with the structure of the graph.

**Comment by mwatts15** [comment]
`Cell` objects represent known cells in a representative C. elegans. A `Cell` object could represent some cell in the C. elegans digestive tract, or the epidermis, or the reproductive organs besides neurons and muscles. Common to all `Cell` objects is the lineage name from which "daughterOf" and "blast" properties are derived.

The utility of subclassing `Neuron` and `Muscle` from `Cell` is limited by the lack of either inclusion of multiple rdf:type relationships, or description of the type hierarchy in the RDF graph together with a query on subtypes of `Cell` along with `Cell`. Adding an rdf:type triple for AVAL marking it as a `Cell` as well as a `Neuron` should produce the expected result for @gsarma's initial example.

Regardless, lineage name properties are included on neurons by insert_worm.py, though not on muscles. The data source used (the Worm Atlas cell table) has a distinct naming convention for muscles from that used in the SQLite database, and I had difficulty finding a reliable mapping between those conventions. IIRC, Wormbase has a mapping if you understand its query language.

**Comment by slarson** [comment]
Thanks for this @mwatts15 

Question:

> Adding an rdf:type triple for AVAL marking it as a Cell as well as a Neuron should produce the expected result for @gsarma's initial example.

How does one accomplish this via code in `insert_worm.py`?  Or perhaps better asked, if someone creates a `Neuron` or `Muscle` in insert_worm.py should we have some code somewhere in the API itself that does this behind the scenes?

**Comment by mwatts15** [comment]
It would be the latter. At one point, I introduced a change that added rdf:type triples for all super-classes of an object's type, but thanks to the way I generated SPARQL queries, this created an order of magnitude slow-down in query execution that was unacceptable.

The introduction of additional type triples can be partially automated by adding n3 rules like:
{ ?x rdf:type :Muscle } => { ?x rdf:type :Cell } .
{ ?x rdf:type :Neuron } => { ?x rdf:type :Cell } .
to the inference engine, assuming the inference engine is in use.
